### ISSUE TITLE
Make the proxy works with html5Mode

### ISSUE THREAD

**Issue by Swiip** [Swiip/generator-gulp-angular#65]
Problem identified here: https://github.com/Swiip/generator-gulp-angular/issues/52#issuecomment-57289375

**Comment by jonlil** [comment]
@raphaelluchini nested urls works fine for me. I updated to latest version yesterday and now my implementation looks like this

```js
'use strict';

var gulp = require('gulp');

var paths = gulp.paths;

var util = require('util');

var browserSync = require('browser-sync');

var modRewrite = require('connect-modrewrite');

function browserSyncInit(baseDir, files, browser) {
  browser = browser === undefined ? 'default' : browser;

  var routes = null;
  if(baseDir === paths.src || (util.isArray(baseDir) && baseDir.indexOf(paths.src) !== -1)) {
    routes = {
      '/bower_components': './bower_components'
    };
  }

  browserSync.instance = browserSync.init(files, {
    startPath: '/',
    server: {
      baseDir: baseDir,
      middleware: modRewrite([
        '!\\.\\w+$ /index.html [L]'
      ]),
      routes: routes
    },
    browser: browser,
    notify: false
  });
}
```

**Comment by raphaelluchini** [comment]
@jonlil I've included <base href="/"> and worked well! ;) Thanks BTW!

**Comment by tuvokki** [comment]
@jonlil thanks, this helped me a lot.

FYI, I use this in combination with a (local) proxy to a backend, like this and that works like a charm:

```javascript
'use strict';

var gulp = require('gulp');
var paths = gulp.paths;
var util = require('util');
var browserSync = require('browser-sync');
var url = require('url'); // https://www.npmjs.org/package/url
var proxy = require('proxy-middleware'); // https://www.npmjs.org/package/proxy-middleware
var modRewrite = require('connect-modrewrite');

function browserSyncInit(baseDir, files, browser) {
  browser = browser === undefined ? 'default' : browser;

  var routes = null;
  if(baseDir === paths.src || (util.isArray(baseDir) && baseDir.indexOf(paths.src) !== -1)) {
    routes = {
      '/bower_components': 'bower_components'
    };
  }

  var proxyOptions = url.parse('http://localhost:9860');
  proxyOptions.route = '/api';
  // requests to `/api/x/y/z` are proxied to `http://localhost:9860`

  browserSync.instance = browserSync.init(files, {
    startPath: '/',
    server: {
      baseDir: baseDir,
      middleware: [proxy(proxyOptions),
        modRewrite([
          '!\\.\\w+$ /index.html [L]'
        ])],
      routes: routes
    },
    browser: browser
  });
}
```

**Comment by zckrs** [comment]
Tested on master branch. No problem

**Comment by uaKorona** [comment]
Hello!  I got a problem with proxy. I need to proxy any request which contains **'mini-account'** redirects to external server http://ws-strob.rhcloud.com/ 

I enabled and did config my proxy.js 
```javascript
var proxy = httpProxy.createProxyServer({
  target: 'http://ws-strob.rhcloud.com/mini-account/'
}).listen(8003);

function proxyMiddleware(req, res, next) {
  if (req.url.indexOf('mini-account') >= 0) {
    console.log('api request: ', req.url);
    proxy.web(req, res);
    //next();
  } else if (/\.(html|css|js|png|jpg|jpeg|gif|ico|xml|rss|txt|eot|svg|ttf|woff|woff2|cur)(\?((r|v|rel|rev)=[\-\.\w]*)?)?$/.test(req.url)) {
    next();
  } else {
    modRewrite(['!\\.\\w+$ /index.html [L]'])(req, res, next);
  }
}
```

and my server.js 
```javascript
 browserSync.instance = browserSync.init(files, {
    startPath: '/',
    server: {
      baseDir: baseDir,
      middleware: middleware,
      routes: routes
    },
    browser: browser
  });
```

I also use html 5 mode in my spa 
```javascript
$urlRouterProvider.otherwise('/');
$locationProvider.html5Mode(true);
```

But when I send reqest to **mini-account/version:8003** 
the proxy redirects me to **https://localhost:3000/mini-account/version:8003** not to external server
**http://ws-strob.rhcloud.com/mini-account/version**

![default](https://cloud.githubusercontent.com/assets/10634202/7805328/b94def38-0378-11e5-90d7-98800d1ace07.png)

Could anyone help me ?

**Comment by marani** [comment]
I guess what you want to do is:
```
var proxy = httpProxy.createProxyServer({
  target: 'http://ws-strob.rhcloud.com:8003/mini-account/'
});
```
or if this doesn't work, try different syntax:
```
var proxy = httpProxy.createProxyServer('/mini-account', {
  target: 'http://ws-strob.rhcloud.com:8003'
});
```
and send your request to `/mini-account/version`

**Comment by zckrs** [comment]
New proxy https://github.com/Swiip/generator-gulp-angular/pull/593 

:tada: Just released v0.12.0 :tada:
Your issue should be fixed. I close but if you can retry your case with a new fresh install of generator you are welcome. Any feedback would be appreciated (positive comments too ^_^).
